### ISSUE TITLE
feat: Use rate limiter when downloading a metadata file

### ISSUE THREAD

**Issue by rdimitrov** [rdimitrov/go-tuf-metadata#27]
Reference: https://github.com/rdimitrov/go-tuf-metadata/blob/e7e2e99d539951126a98f5261e2f00b60f48d210/metadata/fetcher/fetcher.go#L52

**Comment by rdimitrov** [comment]
I purposely left it as a reference as I think it should be sufficient too, but I remember when I tried it there was something weird failing so it needs to be double checked 👍

**Comment by MDr164** [comment]
I just tackled that issue actually and will post the PR in a few minutes. The issue was that the old cold will fail at `strconv.ParseInt` in case the header is not set at all. I solved it by checking if the header was set in the response and if that is not the case the check is skipped. This is no issue as there is now a second check with the `io.LimitReader` I added which reads the unknown length payload at maxLength + 1 and if the data was actually larger than maxLength we return the length error again.
